lib/detect-libc.js

```diff
@@ -77,31 +79,61 @@ const familyFromCommand = (out) => {
  * @returns {Promise<?string>}
  */
 const family = async () => {
-  let family = null;
+  if (cachedFamily !== undefined) {
+    return cachedFamily;
+  }
+  cachedFamily = null;
   if (isLinux()) {
-    family = familyFromReport();
-    if (!family) {
-      const out = await safeCommand();
-      family = familyFromCommand(out);
+    try {
```

Comment:
Please can you wrap this new logic in something like a `familyFromFilesystem()` function, which could manage its own try/catch logic and caching, similar to  `familyFromCommand()` etc.

Reply:
If I do that, I will probably need to have 3 different results, `GLIBC`, `MUSL`, and `FILE_ERROR`.

What do you think?

Reply:
Perhaps return `null` if the family cannot be determined, similar to `familyFromCommand()`, and then try other methods?